src/proposals/20250403/SparkEthereum_20250403.t.sol

```diff
@@ -0,0 +1,457 @@
+
+        _assertRateLimit(depositKey, 0, 0);
+        _assertRateLimit(withdrawKey, 0, 0);
+
+        executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey, 300_000_000e6, 100_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        IERC20 usdc           = IERC20(Ethereum.USDC);
+        ISuperstateToken ustb = ISuperstateToken(Ethereum.USTB);
+
+        // USDS -> USDC limits are 200m, go a bit below in case some is in use
+        uint256 mintAmount = 190_000_000e6;
+        vm.startPrank(ctx.relayer);
+        controller.mintUSDS(mintAmount * 1e12);
+        controller.swapUSDSToUSDC(mintAmount);
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)), mintAmount);
+        assertEq(ustb.balanceOf(address(ctx.proxy)), 0);
+
+        (uint256 ustbShares,,) = ustb.calculateSuperstateTokenOut(mintAmount, address(usdc));
+
+        controller.subscribeSuperstate(mintAmount);
+        vm.stopPrank();
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)), 0);
+        assertEq(ustb.balanceOf(address(ctx.proxy)), ustbShares);
+
+        // Doing a smaller redeem because there is not necessarily enough liquidity
+        vm.prank(ctx.relayer);
+        controller.redeemSuperstate(ustbShares / 100);
+
+        assertApproxEqAbs(usdc.balanceOf(address(ctx.proxy)), mintAmount * 1/100, 100);
+        assertApproxEqAbs(ustb.balanceOf(address(ctx.proxy)), ustbShares * 99/100, 1);
+    }
+
+    function test_ETHEREUM_centrifugeJTRSYOnboarding() public onChain(ChainIdUtils.Ethereum()) {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        
+        MainnetController controller = MainnetController(ETHEREUM_NEW_ALM_CONTROLLER);
+
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_7540_DEPOSIT(),
+            CENTRIFUGE_JTRSY_VAULT
+        );
+        bytes32 withdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_7540_REDEEM(),
+            CENTRIFUGE_JTRSY_VAULT
+        );
+
+        _assertRateLimit(depositKey, 0, 0);
+        _assertRateLimit(withdrawKey, 0, 0);
+
+        executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey, 200_000_000e6, 100_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        IERC20 usdc  = IERC20(Ethereum.USDC);
+        IERC20 jtrsy = IERC20(CENTRIFUGE_JTRSY_TOKEN);
+
+        // USDS -> USDC limits are 200m, go a bit below in case some is in use
+        uint256 mintAmount = 190_000_000e6;
+        vm.startPrank(ctx.relayer);
+        controller.mintUSDS(mintAmount * 1e12);
+        controller.swapUSDSToUSDC(mintAmount);
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)),  mintAmount);
+        assertEq(jtrsy.balanceOf(address(ctx.proxy)), 0);
+
+        controller.requestDepositERC7540(CENTRIFUGE_JTRSY_VAULT, mintAmount);
+        vm.stopPrank();
+
```

Comment:
Need rate limit assertions after requestDeposit and requestRedeem

Reply:
added